docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+Microsoft Research. That model is based on having a single copy from the replication group that is acting as the primary shard.
+The other copies are called “replica shards”. The primary serves as the main entry point for all indexing operations. It is in charge of
+validating them and making sure they are correct. Once an index operation has been accepted by the primary, the primary is also
+responsible for replicating the operation to the other copies.
+
+This purpose of this section is to give a high level overview of the Elasticsearch replication model and discuss the implications
+it has for various interactions between write and read operations.
+
+=== Basic write model
+
+Every indexing operation in Elasticsearch is first resolved to a replication group.
+This is <<index-routing,typically>> done by looking at the document ID. Once the replication group is determined,
+the operation is internally forwarded to the current primary shard of the group. The primary shard is responsible
+for validating the operation and forwarding it to the other replicas. Since replicas can be off-line, the primary
+is not required to replicate to all replicas. Instead, Elasticsearch maintains a list of shard copies that should
+receive the operation. This list is called the “in-sync copies” and is maintained by the master node. As the name implies,
+these are the set of "good" shard copies that are guaranteed to have processed all of the index and delete operations that
+have been acknowledged to the user. The primary is responsible for maintaining this invariant and thus has to replicate all
+operations to each copy in this set.
+
+The primary shard follows this basic flow:
+
+. Validate incoming operation and reject it if structurally invalid (Example: have an object field where a number is expected)
+. Execute the operation locally - i.e., indexing or deleting the relevant document. This will also validate the content of fields
+   and reject if needed (Example: a keyword value is too long for indexing in Lucene).
+. Forward the operation to each replica in the current in-sync copies set. If there are multiple replicas, this is done concurrently.
+. Once all replicas have successfully performed the operation and responded to the primary, the primary acknowledges the successful
+   completion of the request to the client.
+
+==== Failure handling
+
+Many things can go wrong during indexing - disks can get corrupted, nodes can be disconnected from each other or some
+configuration mistake could cause an operation to fail on a replica despite it being successful on the primary. These
+are infrequent but the primary has to respond to them.
+
+In the case that the primary itself fails, a message will be sent to the master about it. The indexing operation will
```

Comment:
a message will be sent -> who sends the message?  I'd prefer to name the sender and to use the active voice instead of passive here